From ff7122747386e73f440994a25b966f774259dd06 Mon Sep 17 00:00:00 2001
From: Tobias Schramm <t.schramm@t-sys.eu>
Date: Mon, 22 Mar 2021 00:49:54 +0100
Subject: [PATCH 10/11] ASoC: soc: codec: tas5825m: make gpiomux table const

Signed-off-by: Tobias Schramm <t.schramm@t-sys.eu>
---
 sound/soc/codecs/tas5825m.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/codecs/tas5825m.c b/sound/soc/codecs/tas5825m.c
index ff4f13d04b41..314410f78042 100644
--- a/sound/soc/codecs/tas5825m.c
+++ b/sound/soc/codecs/tas5825m.c
@@ -630,7 +630,7 @@ struct tas5825m_gpio_func {
 	unsigned mux;
 };
 
-struct tas5825m_gpio_func tas5825m_gpio_fuctions[] = {
+const struct tas5825m_gpio_func tas5825m_gpio_fuctions[] = {
 	{ "gpio",		0b0010 },
 	{ "auto-mute",		0b0011 },
 	{ "auto-mute-left",	0b0100 },
@@ -647,7 +647,7 @@ static int tas5825m_get_gpio_func_by_name(const char *name) {
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(tas5825m_gpio_fuctions); i++) {
-		struct tas5825m_gpio_func *func = &tas5825m_gpio_fuctions[i];
+		const struct tas5825m_gpio_func *func = &tas5825m_gpio_fuctions[i];
 
 		if (!strcmp(name, func->name)) {
 			return func->mux;
-- 
2.30.1

